editor/editor_help.cpp

```diff
@@ -2620,6 +2621,166 @@ void EditorHelpBit::_meta_clicked(String p_select) {
 	}
 }
 
+String EditorHelpBit::get_class_description(const StringName &p_class_name) const {
+	HashMap<String, DocData::ClassDoc>::ConstIterator E = EditorHelp::get_doc_data()->class_list.find(p_class_name);
+	if (!E) {
+		return String();
+	}
+
+	// Non-native properties shouldn't be translated.
+	bool is_native = ClassDB::class_exists(p_class_name);
+	// Caching descriptions would be overkill.
+	return is_native ? DTR(E->value.brief_description) : E->value.brief_description;
+}
+
+String EditorHelpBit::get_property_description(const StringName &p_class_name, const StringName &p_property_name) const {
+	HashMap<String, DocData::ClassDoc>::ConstIterator E = EditorHelp::get_doc_data()->class_list.find(p_class_name);
+	if (!E) {
+		return String();
+	}
+
+	String description;
+	if (doc_property_cache.has(p_class_name) && doc_property_cache[p_class_name].has(p_property_name)) {
+		description = doc_property_cache[p_class_name][p_property_name];
+	} else {
+		// Non-native properties shouldn't be cached, nor translated.
+		bool is_native = ClassDB::class_exists(p_class_name);
+
+		for (int i = 0; i < E->value.properties.size(); i++) {
+			String description_current = is_native ? DTR(E->value.properties[i].description) : E->value.properties[i].description;
+
+			const Vector<String> class_enum = E->value.properties[i].enumeration.split(".");
+			const String enum_name = class_enum.size() >= 2 ? class_enum[1] : "";
+			if (!enum_name.is_empty()) {
+				for (DocData::ConstantDoc val : E->value.constants) {
```

Comment:
This is not equivalent. The old code in the inspector was taking the class name from the enum as well, because enum types can be from another class. So checking only the current class is not sufficient.

Reply:
Can you point me to a example so I can test this myself?

Reply:
Button's text overrun behavior is the first example I could find:

**master**
<img width="366" alt="godot windows editor dev x86_64_2023-09-29_13-34-33" src="https://github.com/godotengine/godot/assets/11782833/c60b0da7-9773-4ee8-9347-707811f931a4">

**This PR**
<img width="397" alt="godot windows editor dev x86_64_2023-09-29_13-30-04" src="https://github.com/godotengine/godot/assets/11782833/2c027047-61bb-46e3-80e7-12f61a0ced21">

Unrelated to this PR, but `alignment` just above it doesn't show the list of options, I assume because this feature doesn't work with global scope enums for some reason. We should fix it later.